### PR TITLE
`eth_getCode` to return the redirect bytecode for HTS tokens

### LINKED ISSUE #583: HTS NFTs not supported in Metamask mobile wallet
### Problem

HTS NFTs are not supported by Metamask mobile app. When attempting to add an HTS NFT, the metamask mobile wallet errors with `Personal address detected. Input the collectible contract address`

image

Note: User is able to add the same NFT Collection (ie token_id) to the Metamask chrome extension. 

Details:
Platform: Hedera testnet
token_id: 0.0.48014385
evmAddress: `0x0000000000000000000000000000000002dca431`
OS: Android 13 
MM version: 5.7.0 (972)

Steps to reproduce: 
1. Create HTS NFT and send to your account
2. Open MM mobile wallet select "NFTs"
3. Click "import NFTs" 
4. Enter Address and ID of the NFT which is already in your account. 
5. Note error message `Personal address detected. Input the collectible contract address`

To test ERC721 tokens, do the exact same steps with an ERC721 asset in your account. 

### Solution

So on MM mobile, it specifically asks for the contract address and ID whereas the chrome extension does *NOT* ask for ID. The ID here is the serial number of the NFT. 

I have no issues adding a Hedera ERC721 NFT asset to MM mobile, only HTS NFTs. 

### Alternatives

_No response_

### LINKED ISSUE #594: `eth_getCode` to return the redirect bytecode for HTS tokens
### Description

As reported here: https://github.com/hashgraph/hedera-json-rpc-relay/issues/583 when we perfrom eth_getCode for HTS tokens, currently the result for HTS token addresses is INVALID_CONTRACT_ID as we do not return the redirect bytecode. 

As described in the issue logged in the consensus node: https://github.com/hashgraph/hedera-services/issues/3659
Although  a consensus node change could be made, it will be more cost efficient (hbar cost) if the contract bytecode call to the mirror node fails to then see if it is a HTS token (checking mirror node) and then apply the proxy code to it. Similar network load, but the GetContractBytecode call will cost on the first hit if not cached.

### Steps to reproduce

Described in an example here: https://github.com/hashgraph/hedera-json-rpc-relay/issues/583 

### Additional context

Redirect bytecode: 
```
6080604052348015600f57600080fd5b506000610167905077618dc65efefefefefefefefefefefefefefefefefefefefe600052366000602037600080366018016008845af43d806000803e8160008114605857816000f35b816000fdfea2646970667358221220d8378feed472ba49a0005514ef7087017f707b45fb9bf56bb81bb93ff19a238b64736f6c634300080b0033
```

`fefefefefefefefefefefefefefefefefefefefe` should be replaced with token address
### Hedera network

mainnet

### Version

v0.8.0

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:

- Expand `resolveEntityType` function in mirror node to support token types
- Change logic in `eth_getCode` - check for address type, if it's HTS token return redirect bytecode, rest is the same as before
- `address` to `tokenId` conversion by using `parseInt(address, 16)`
- Added acceptance test to cover this use case
- Added unit test to cover this use case

**Related issue(s)**:

Fixes #594 
Fixes #583 

**Notes for reviewer**:
Little change in unit tests, as we are making some additional requests and we need to mock them.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/clients/mirrorNodeClient.ts [hunk 1]
```diff
@@ -356,6 +357,13 @@ export class MirrorNodeClient {
             .replace(MirrorNodeClient.TIMESTAMP_PLACEHOLDER, timestamp);
     }
 
+    public async getTokenById(tokenId: string, requestId?: string) {
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply packages/relay/src/lib/clients/mirrorNodeClient.ts:360]: Done.

File: packages/relay/src/lib/eth.ts [hunk 2]
```diff
@@ -1015,7 +1015,9 @@ export class EthImpl implements Eth {
   }
 
   private static redirectBytecodeAddressReplace(address: string): string {
-    return `6080604052348015600f57600080fd5b506000610167905077618dc65e${address.slice(2)}600052366000602037600080366018016008845af43d806000803e8160008114605857816000f35b816000fdfea2646970667358221220d8378feed472ba49a0005514ef7087017f707b45fb9bf56bb81bb93ff19a238b64736f6c634300080b0033`
+    const redirectBytecodePrefix = '6080604052348015600f57600080fd5b506000610167905077618dc65e';
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:1018]: nit: these should be class `const`

**Review by Nana-EC — APPROVED** [review]
LG

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**